tests/test_utils.py

```diff
@@ -108,3 +113,45 @@ def test_build_dict(list_for_dict):
     assert dict_from_list['John']['age'] == 20
     assert dict_from_list['Mark']['age'] == 27
     assert dict_from_list['Bruce']['age'] == 9
+
+
+def test_split_fullname():
+    """Test author fullname splitting."""
+    author1 = 'Doe, John Magic'
+    author2 = 'Doe Boe, John Magic'
+    author3 = 'Doe Boe John Magic'
+    author4 = 'John Magic Doe'
+    author5 = 'John Magic Doe Boe'
+    author6 = 'John Magic, Doe Boe'
+    assert split_fullname(author1) == ('Doe', 'John Magic')
+    assert split_fullname(author2) == ('Doe Boe', 'John Magic')
+    assert split_fullname(author3) == ('Doe', 'Boe John Magic')
+    assert split_fullname(author4, surname='last') == ('Doe', 'John Magic')
+    assert split_fullname(author5, surname='last') == ('Boe', 'John Magic Doe')
+    assert split_fullname(author6, surname='last') == ('Doe Boe', 'John Magic') 
+
+
+def test_parse_domain():
+    """Test domain parsing."""
+    url = 'http://www.example.com/0351/content'
+    domain = 'http://www.example.com/'
+    assert parse_domain(url) == domain
+
+
+@responses.activate
+def test_get_mime_type():
+    """Test MIME type getting.
+
+    Apparently the mocked response always has 'Content-Type'
+    and it seems to default to 'text/html'?
```

Comment:
Yeah, or `text/plain`, what you can do is to add another test function where in your mock you try with `content_type=None` as part of the arguments.